frontend/src/metabase/lib/formatting/time.ts

```diff
@@ -67,3 +67,34 @@ export function formatTimeWithUnit(
 
   return m.format(timeFormat);
 }
+
+interface TimeOnlyOptions {
+  local?: boolean;
+  time_enabled?: "minutes" | "milliseconds" | "seconds" | null;
+  time_format?: string;
+  time_style?: string;
+}
+
+export function formatTimeWithOptions(
+  time: Moment,
+  unit: DatetimeUnit,
+  options: TimeOnlyOptions = {},
+) {
+  const parsedTime = parseTime(time);
+
+  const timeStyle = options.time_style
+    ? options.time_style
+    : DEFAULT_TIME_STYLE;
+
+  const timeEnabled = options.time_enabled
```

Comment:
Nit: I thought we had a lint rule against nested ternaries since they are hard to read but I guess not. An if-else would be easier to read.

```typescript
  let timeEnabled;
  if (options.time_enabled) {
    timeEnabled = options.time_enabled;
  } else if (hasHour(unit)) {
    timeEnabled = "minute";
  } else {
    timeEnabled = null;
  }
  ```